src/shared/components/download-button/DownloadButton.scss

```diff
@@ -18,3 +18,45 @@
   --download-button-background: #{$grey};
   cursor: default;
 }
+
+.downloadButtonLoading {
+  background-color: var(--download-button-loading-background, transparent);
+  --circle-loader-diameter: var(--_download-button-size);
+  transform: scale(1.2);
+}
+
+.downloadButtonLoading,
+.downloadButtonSuccess,
+.downloadButtonError {
+  cursor: default;
+}
+
+.downloadButtonSuccess,
+.downloadButtonError {
+  background-color: var(--download-button-success-background, transparent);
+  border: none;
+  display: inline-flex;
+  align-items: center;
+  justify-content: center;
+
+  svg {
+    height: auto;
+  }
```

Comment:
Are all these styles needed?

Reply:
Checked again. `border:none` not required. Display inline-flex with the centering of the child isn't really necessary for the success button, but is needed for the error button, because the cross in the error state is smaller than the size of the button. I will leave the inline-flex for both success and error to avoid confusion.

✅